tree/ntuple/v7/test/ntuple_merger.cxx

```diff
@@ -1070,3 +1070,124 @@ TEST(RNTupleMerger, MergeCompression)
+      RNTupleMerger merger;
+      {
+         auto destination = std::make_unique<RPageSinkFile>("ntuple", fileGuard3.GetPath(), RNTupleWriteOptions());
+         auto opts = RNTupleMergeOptions();
+         opts.fCompressionSettings = 0;
+         auto res = merger.Merge(sourcePtrs, *destination, opts);
+         // TODO(gparolini): we want to support this in the future
+         EXPECT_FALSE(bool(res));
+         if (res.GetError())
+            EXPECT_THAT(res.GetError()->GetReport(), testing::HasSubstr("different column type"));
+         // EXPECT_TRUE(bool(res));
+      }
+      {
+         auto destination = std::make_unique<RPageSinkFile>("ntuple", fileGuard4.GetPath(), RNTupleWriteOptions());
+         auto res = merger.Merge(sourcePtrs, *destination);
+         // TODO(gparolini): we want to support this in the future
+         EXPECT_FALSE(bool(res));
+         if (res.GetError())
+            EXPECT_THAT(res.GetError()->GetReport(), testing::HasSubstr("different column type"));
+         // EXPECT_TRUE(bool(res));
+      }
+   }
+}
+
+TEST(RNTupleMerger, MultipleRepresentations)
+{
+   // verify that we properly handle ntuples with multiple column representations
+   FileRaii fileGuard1("test_ntuple_merge_multirep_in_1.root");
+
+   {
+      auto model = RNTupleModel::Create();
+      auto fldPx = RFieldBase::Create("px", "float").Unwrap();
+      fldPx->SetColumnRepresentatives({{EColumnType::kReal32}, {EColumnType::kReal16}});
+      model->AddField(std::move(fldPx));
+      auto ptrPx = model->GetDefaultEntry().GetPtr<float>("px");
+      auto writer = RNTupleWriter::Recreate(std::move(model), "ntuple", fileGuard1.GetPath());
+      *ptrPx = 1.0;
+      writer->Fill();
+      writer->CommitCluster();
+      ROOT::Experimental::Internal::RFieldRepresentationModifier::SetPrimaryColumnRepresentation(
+         const_cast<RFieldBase &>(writer->GetModel().GetConstField("px")), 1);
+      *ptrPx = 2.0;
+      writer->Fill();
+   }
+
+   // Now merge the inputs
+   FileRaii fileGuard2("test_ntuple_merge_multirep_out.root");
+   {
+      // Gather the input sources
+      std::vector<std::unique_ptr<RPageSource>> sources;
+      sources.push_back(RPageSource::Create("ntuple", fileGuard1.GetPath()));
+      sources.push_back(RPageSource::Create("ntuple", fileGuard1.GetPath()));
+      std::vector<RPageSource *> sourcePtrs;
+      for (const auto &s : sources) {
+         sourcePtrs.push_back(s.get());
+      }
+
+      auto sourcePtrs2 = sourcePtrs;
+
+      RNTupleMerger merger;
+      {
+         auto destination = std::make_unique<RPageSinkFile>("ntuple", fileGuard2.GetPath(), RNTupleWriteOptions());
+         auto opts = RNTupleMergeOptions();
+         opts.fCompressionSettings = 0;
+         auto res = merger.Merge(sourcePtrs, *destination, opts);
+         // TODO(gparolini): we want to support this in the future
+         // XXX: this currently fails because of a mismatch in the number of columns of dst vs src.
```

Comment:
That's a good point! I think that is an oversight in `RNTupleDescriptor::CreateModel()`.

Reply:
Well... let's discuss tomorrow. For the moment I agree, the important point is that this case fails gracefully.